packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx

```diff
@@ -503,15 +506,113 @@ const PublishAction: DocumentActionComponent = ({
   const isSubmitting = useForm('PublishAction', ({ isSubmitting }) => isSubmitting);
   const validate = useForm('PublishAction', (state) => state.validate);
   const setErrors = useForm('PublishAction', (state) => state.setErrors);
   const formValues = useForm('PublishAction', ({ values }) => values);
 
+  React.useEffect(() => {
+    if (isErrorDraftRelations) {
+      toggleNotification({
+        type: 'danger',
+        message: formatMessage({
+          id: getTranslation('error.records.fetch-draft-relatons'),
+          defaultMessage: 'An error occurred while fetching draft relations on this document.',
+        }),
+      });
+    }
+  }, [isErrorDraftRelations, toggleNotification, formatMessage]);
+
+  React.useEffect(() => {
+    let totalDraftRelations = 0;
+    const localDraftRelations = new Set();
+
+    /**
+     * Extracts draft relations from the provided data object.
+     * It checks for a connect array of relations.
+     * If a relation has a status of 'draft', its id is added to the localDraftRelations set.
+     */
+    const extractDraftRelations = (data: Omit<RelationsFormValue, 'disconnect'>) => {
+      const relations = data.connect || [];
+      relations.forEach((relation) => {
+        if (relation.status === 'draft') {
+          localDraftRelations.add(relation.id);
+        }
+      });
+    };
+
+    /**
+     * Recursively traverses the provided data object to extract draft relations from arrays within 'connect' keys.
+     * If the data is an object, it looks for 'connect' keys to pass their array values to extractDraftRelations.
+     * It recursively calls itself for any non-null objects it contains.
+     */
+    const traverseAndExtract = (data: { [field: string]: any }) => {
+      Object.entries(data).forEach(([key, value]) => {
+        if (key === 'connect' && Array.isArray(value)) {
+          extractDraftRelations({ connect: value });
+        } else if (typeof value === 'object' && value !== null) {
+          traverseAndExtract(value);
+        }
+      });
+    };
+
+    if (!documentId || modified) {
+      traverseAndExtract(formValues);
+
+      totalDraftRelations += localDraftRelations.size;
+    }
+
+    if (!documentId) {
+      // If the document does not yet exist, we don't need to check for draft
+      // relations on the backend.
+
+      setCountOfDraftRelations(totalDraftRelations);
+      return;
+    }
+
+    if (!isLoadingDraftRelations) {
+      const checkForDraftRelations = async () => {
+        const { data, error } = await countDraftRelations({
+          collectionType,
+          model,
+          documentId,
+          params,
+        });
+
+        if (error) {
+          throw error;
+        }
+
+        if (data) {
+          totalDraftRelations += data.data;
+          setCountOfDraftRelations(totalDraftRelations);
+        }
+      };
+
+      checkForDraftRelations();
+    }
```

Comment:
when does this effect run? Is it every time the form values change? or when the user tries to publish an entry?

Even with this if (!isLoadingDraftRelations) check, if this runs on every keypress on the edit view form, we might be spamming the server with many requests that migth not be necessary